src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWallet.scala

```diff
@@ -32,7 +32,7 @@ class ErgoWallet(historyReader: ErgoHistoryReader, settings: ErgoSettings)
   override type NVCT = this.type
 
   override val walletActor: ActorRef = {
-    val props = Props(classOf[ErgoWalletActor], settings, boxSelector, historyReader)
+    val props = Props(classOf[ErgoWalletActor], settings, new ErgoWalletServiceImpl, boxSelector, historyReader)
```

Comment:
`ErgoWalletService` could be just `object` as it has no state, but this way we can mock it and test the logic of the Actor, there is still some business logic, for instance Recursive scanning that must be done by message passing to not block the actor ...